samcli/lib/build/app_builder.py

```diff
@@ -187,17 +188,24 @@ def _build_layer(self, layer_name, codeuri, runtime):
             manifest_path = self._manifest_path_override or os.path.join(code_dir, config.manifest_name)
 
             # By default prefer to build in-process for speed
+            build_runtime = specified_workflow
             build_method = self._build_function_in_process
             if self._container_manager:
-                build_method = self._build_function_on_container
+                build_method = self._build_function_in_process
+                if config.language == "provided":
+                    LOG.warning(
+                        "For container layer build, first compatible runtime is chosen as build target for container.")
+                    # Only set to this value if specified workflow is makefile which will result in config language as provided
+                    build_runtime = compatible_runtimes[0]
```

Comment:
If no compatible runtimes are specified, Im hoping the transform itself would fail before we get to this code path, so we should be good?

Reply:
Yeah, with no compatible runtime we don't reach here.